render/egl.c

```diff
@@ -413,6 +416,40 @@ EGLImageKHR wlr_egl_create_image_from_dmabuf(struct wlr_egl *egl,
 		EGL_LINUX_DMA_BUF_EXT, NULL, attribs);
 }
 
+bool wlr_egl_export_image_to_dmabuf(struct wlr_egl *egl, EGLImageKHR image,
+		int32_t width, int32_t height, uint32_t flags,
+		struct wlr_dmabuf_buffer_attribs *attribs) {
+	memset(attribs, 0, sizeof(struct wlr_dmabuf_buffer_attribs));
+
+	if (!egl->egl_exts.dmabuf_export || !eglExportDMABUFImageQueryMESA ||
+			!eglExportDMABUFImageMESA) {
+		return false;
+	}
+
+	EGLuint64KHR modifier;
+	if (!eglExportDMABUFImageQueryMESA(egl->display, image,
+			(int *)&attribs->format, &attribs->n_planes, &modifier)) {
+		return false;
+	}
+	if (attribs->n_planes > WLR_LINUX_DMABUF_MAX_PLANES) {
+		wlr_log(L_ERROR, "EGL returned more planes than supported");
+		return false;
+	}
+	for (int i = 0; i < attribs->n_planes; ++i) {
+		attribs->modifier[i] = modifier;
```

Comment:
Need to copy per plane modifiers here (rather than using the one for the first plane for all planes).

Reply:
```
attribs->modifier[i] = modifiers[i];
```

Reply:
How do I know the length of the array to pass to `eglExportDMABUFImageQueryMESA`?

Reply:
Rereading the mesa code I'm plain wrong here. It's returning the supported modifiers. Scratch all that and sorry for the confusion.